services/fee-estimator/shared/dynamicFees.js

```diff
@@ -34,14 +34,25 @@ const calculateBlockSize = async block => {
 };
 
 const calculateWeightedAvg = async blocks => {
-	const blockSizes = await Promise.all(blocks.map(block => calculateBlockSize(block)));
-	const decayFactor = config.feeEstimates.wavgDecayPercentage / 100;
+	const blockSizes = await BluebirdPromise.map(
+		blocks,
+		async block => calculateBlockSize(block),
+		{ concurrency: blocks.length },
+	);
+	const decayFactor = 1 - (config.feeEstimates.wavgDecayPercentage / 100);
 	let weight = 1;
-	const wavgLastBlocks = blockSizes.reduce((a = 0, b = 0) => {
-		weight *= 1 - decayFactor;
-		return a + (b * weight);
-	});
+	let totalWeight = 0;
+
+	const wSumLastBlocks = blockSizes.reduce(
```

Comment:
Personal taste: I would use no abbreviation in variables like this and rather write `weightedSum` just o clarify the code as much as possible

Reply:
In the same spirit, I find this reduce function a bit confusing: `wBlockAcc` is returned, while `weight` is updated in place. Anyway, it looks okay to me after reading the reduce docs, but it would be nice to try the following case:   array of 20 blocks all with the same size. In this case the avg should equal the size of the blocks.

Reply:
@priojeetpriyom Could you please address the above points in the PR?

Reply:
The pr is addressed. Thanks for the test suggestion @ricott1 🥇